### PR TITLE
[docs] [resources-config-overhaul] - Update Job Execution Concept page

### PR DESCRIPTION
**Pull request by yuhan**
## Summary & Motivation
update execution config. but in addition to https://github.com/dagster-io/dagster/pull/13696, i think we also need to expose something like `MultiprocessConfig`

### CODE REVIEW

**Comment by yuhan** [comment]
Current dependencies on/for this PR:
* master
  * **PR #13704** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13704" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/13704?utm_source=stack-comment).

File: docs/content/concepts/ops-jobs-graphs/job-execution.mdx [hunk 1]
```diff
@@ -196,23 +199,26 @@ Limits can be specified for all ops with a certain tag key or key-value pair. If
...
                     "tag_concurrency_limits": [
                         {
                             "key": "database",
                             "value": "redshift",
                             "limit": 2,
                         }
                     ],
-                },
+                }
             }
```
Comment (schrockn) [inline docs/content/concepts/ops-jobs-graphs/job-execution.mdx:219]: @benpankow  we should be able to support strongly typed config on executors

**Review by schrockn — CHANGES_REQUESTED** [review]
I think we should wait to land until we can type this whole thing with `DefaultExecutorConfig`

File: docs/content/guides/limiting-concurrency-in-data-pipelines.mdx [hunk 2]
```diff
@@ -255,23 +255,26 @@ assets_job = define_asset_job(
...
+
 @job(
-    config={
-        "execution": {
+    config=RunConfig(
+        execution={
             "config": {
                 "multiprocess": {
-                    "max_concurrent": 4,
+                    "max_concurrent": 0,
```
Comment (erinkcochran87) [inline docs/content/guides/limiting-concurrency-in-data-pipelines.mdx:266]: Same comment here as on job execution - this change puts the example at odds with the paragraph that comes before it.

**Comment by yuhan** [comment]
closing - too old to pick up. will start a new branch

**Pull request closed without merging**